modules/percolator/src/main/java/org/opensearch/percolator/PercolateQueryBuilder.java

```diff
@@ -279,16 +206,38 @@ public PercolateQueryBuilder(
         this.documentSupplier = null;
     }
 
+    protected PercolateQueryBuilder(String field, Supplier<BytesReference> documentSupplier) {
+        if (field == null) {
+            throw new IllegalArgumentException("[field] is a required argument");
+        }
+        this.field = field;
+        this.documents = Collections.emptyList();
+        this.documentXContentType = null;
+        this.documentSupplier = documentSupplier;
+        indexedDocumentIndex = null;
+        indexedDocumentId = null;
+        indexedDocumentRouting = null;
+        indexedDocumentPreference = null;
+        indexedDocumentVersion = null;
+    }
+
     /**
      * Read from a stream.
      */
     PercolateQueryBuilder(StreamInput in) throws IOException {
         super(in);
         field = in.readString();
         name = in.readOptionalString();
-        documentType = in.readOptionalString();
+        if (in.getVersion().before(Version.V_2_0_0)) {
+            String documentType = in.readOptionalString();
+            assert documentType == null;
+        }
         indexedDocumentIndex = in.readOptionalString();
-        indexedDocumentType = in.readOptionalString();
+        if (in.getVersion().before(Version.V_2_0_0)) {
+            String indexedDocumentType = in.readOptionalString();
+            assert indexedDocumentType == null;
```

Comment:
Is this assertion needed? `readOptionalString` assumes `null` value but not necessarily

Reply:
Thank you @reta for the comment. Typeless version of this query in 7.x is represented by a null documentType and indexedDocumentType.

Reply:
Instead of assert do we want to throw `IllegalArgumentException` since assertions could be disabled? (this is why we did it in other PRs so it would be consistent with those serialization methods)

Reply:
Thanks @nknize for the comment. I will replace assertion with throw `IllegalArgumentException`.